akka-docs/src/main/paradox/typed/mailboxes.md

```diff
@@ -37,4 +39,180 @@ Java
 
 @@snip [MailboxDocSpec.scala](/akka-actor-typed-tests/src/test/resources/mailbox-config-sample.conf) { }
 
-For more details on advanced mailbox config and custom mailbox implementations, see @ref[Classic Mailboxes](../mailboxes.md#builtin-mailbox-implementations).
+### Requiring a Message Queue Type for a Dispatcher
+
+A dispatcher may also have a requirement for the mailbox type used by the
+actors running on it. An example is the @apidoc[BalancingDispatcher] which requires a
+message queue that is thread-safe for multiple concurrent consumers. Such a
+requirement is formulated within the dispatcher configuration section like
+this:
+
+```
+my-dispatcher {
+  mailbox-requirement = org.example.MyInterface
+}
+```
+
+The given requirement names a class or interface which will then be ensured to
+be a supertype of the message queue’s implementation. In case of a
+conflict—e.g. if the actor requires a mailbox type which does not satisfy this
+requirement—then actor creation will fail.
+
+### How the Mailbox Type is Selected
+
+When an actor is created, the `ActorRefProvider` first determines the
+dispatcher which will execute it. Then the mailbox is determined as follows:
+
+ 1. If the actor’s deployment configuration section contains a `mailbox` key
+then that names a configuration section describing the mailbox type to be
+used.
```

Comment:
The way this is written is confusing to me, can it be clarified - the sentence structure?